build.assets/Dockerfile

```diff
@@ -291,9 +291,8 @@ RUN helm plugin install https://github.com/vbehar/helm3-unittest && \
     HELM_PLUGINS=/home/ci/.local/share/helm/plugins helm plugin list
 
 # Install JS gRPC tools.
-ARG NODE_GRPC_TOOLS_VERSION # eg, "1.12.4"
-ARG NODE_PROTOC_TS_VERSION  # eg, "5.0.1"
-RUN npm install --global "grpc-tools@$NODE_GRPC_TOOLS_VERSION" "grpc_tools_node_protoc_ts@$NODE_PROTOC_TS_VERSION"
+ARG PROTOBUF_TS_PLUGIN_VERSION  # eg, "2.9.3"
+RUN npm install --global "@protobuf-ts/plugin@$PROTOBUF_TS_PLUGIN_VERSION"
```

Comment:
I'm not sure if we won't have to do this in three steps, where we first add `@protobuf-ts/plugin` to the Dockerfile, then add code that uses it and then remove `grpc-tools` and `grpc_tools_node_protoc_ts`.

I'm not up to date how the GitHub Actions work nowadays, but with GCP I remember issues with the fact that the builders were using cached/published version of the Dockerfile and not the one from the branch.

The lint action failed on this:

```
buf generate --template=buf-ts.gen.yaml --path=proto/prehog/ --path=proto/teleport/lib/teleterm/ --path=api/proto/teleport/userpreferences/ --path=api/proto/teleport/accesslist/ --path=api/proto/teleport/trait/ --path=api/proto/teleport/header/
Failure: plugin typescript: exec: "/usr/local/lib/node_modules/@protobuf-ts/plugin/bin/protoc-gen-ts": stat /usr/local/lib/node_modules/@protobuf-ts/plugin/bin/protoc-gen-ts: no such file or directory
```

https://github.com/gravitational/teleport/actions/runs/7585240077/job/20660640230?pr=36921#step:15:17

But I'm not 100% sure if it was caused by what I'm talking about.

Reply:
Yeah I think you're correct. Let me split out this part into a new PR

Reply:
https://github.com/gravitational/teleport/pull/37009